source/docs/v3/tenancy-bootstrappers.blade.md

```diff
@@ -29,31 +29,76 @@ php artisan cache:clear --tag=tenant_123
-    - Because the `asset()` helper uses the TenantAssetController to retrieve tenant-specific files. Be aware this requires the tenant to be identified and initialized to function. Accordingly, you should ensure that the tenant identification middleware used is appropriate for your use case (defaults to InitializeTenancyByDomain). For example, if you are only using subdomain identification, you should update the middleware used by the TenantAssetController to the InitializeTenancyBySubdomain middleware. To do this, you can follow the steps outlined in [configuration]({{ $page->link('configuration') }}) to update the public static property which defined the middleware used, e.g. in your TenancyServiceProvider:
-    
 ```php
-use Stancl\Tenancy\Controllers\TenantAssetsController;
-use Stancl\Tenancy\Middleware\InitializeTenancyByDomainOrSubdomain;
+// Tenancy config (tenancy.filesystem.root_override)
+// %storage_path% gets replaced by storage_path()'s output
+// E.g. Storage::disk('local')->path('') will return "/$path_to_your_application/storage/$suffix/app"
+'root_override' => [
+    'local' => '%storage_path%/app/',
+    'public' => '%storage_path%/app/public/',
+],
+```
 
-    // other methods in the TenancyServiceProvider
+To make the tenant-aware `Storage` facade work with a custom disk, add the disk's name to `config('tenancy.filesystem.disks')` and override its root in `config('tenancy.filesystem.root_override')` as shown above.
 
-    public function boot()
-    {
-        // update the middleware used by the asset controller
-        TenantAssetsController::$tenancyMiddleware = InitializeTenancyByDomainOrSubdomain::class;
-    }
+### Assets
+
+The bootstrapper modifies the links retrieved by the `asset()` helper, so they link to the files *of the currently initialized and identified tenant*.
+
+Before using the `asset()` helper, make sure to [assign the identification middleware you're using in your app to TenantAssetsController's `$tenancyMiddleware`]({{ $page->link('configuration#static-properties') }}):
+
+```php
+// TenancyServiceProvider (don't forget to import the classes)
+
+public function boot()
+{
+    // Update the middleware used by the asset controller
+    TenantAssetsController::$tenancyMiddleware = InitializeTenancyByDomainOrSubdomain::class;
+}
 ```
 
-This bootstrapper is the most complex one, by far. We will have a — better written — explanation in v3 docs soon, but for now, refer to the 2.x docs for information about filesystem tenancy. [https://tenancyforlaravel.com/docs/v2/filesystem-tenancy/](https://tenancyforlaravel.com/docs/v2/filesystem-tenancy/)
+> Note: You can disable tenancy of `asset()` in the config (`tenancy.filesystem.asset_helper_tenancy`) and explicitly use `tenant_asset()` instead. You may want to do that if you're facing issues using a package that utilizes `asset()` inside the tenant app.
```

Comment:
Do `asset()` and `tenant_asset()` work interchangeably? When asset helper tenancy is enabled, that is.

Reply:
I just read @abrardev99's comments about the helpers, and my bad for not trying this out myself, I was under the impression that they had the same output when asset helper tenancy was enabled. 

Responded on BC: https://3.basecamp.com/5170965/buckets/23651018/todos/4796489363#__recording_5347808448

Reply:
> I was under the impression that they had the same output when asset helper tenancy was enabled.

Same, that's why I was confused about the BC thread.

Reply:
Should I update this now, or should we wait for the `tenant_asset()` update?

Reply:
You're writing v3 docs, v4 changes don't have an impact on this. Or what do you mean by update?

Reply:
> Or what do you mean by update?

I meant making `tenant_asset()` work as we expected it to work. You're implying that'll be done in v4, so I'll update the docs to match how it works in v3.

Reply:
Updated

Reply:
We won't be making any changes to `tenant_asset()` in v3, way too many breaking changes could come out of that.